tests/python/gaia-ui-tests/gaiatest/apps/messages/app.py

```diff
@@ -52,20 +64,57 @@ def tap_first_received_message(self):
         from gaiatest.apps.messages.regions.message_thread import MessageThread
         return MessageThread(self.marionette)
 
+    def tap_thread_options(self):
+        self.marionette.find_element(*self._thread_options_button).tap()
+        from gaiatest.apps.messages.regions.options import ThreadOptions
+        return ThreadOptions(self.marionette)
+
+    def enter_select_mode(self):
+        edit_header = self.marionette.find_element(*self._edit_thread_header)
+        thread_options = self.tap_thread_options()
+        thread_options.select_threads()
+        Wait(self.marionette).until(lambda m: edit_header.rect['y'] == 0)
```

Comment:
rect calls getElementRect in the marionette listener, I think, which uses getBoundingClientRect()
http://mxr.mozilla.org/mozilla-central/source/testing/marionette/driver/marionette_driver/marionette.py#149
http://mxr.mozilla.org/mozilla-central/source/testing/marionette/listener.js#1569
When an element is display: none, the .x and .y values are 0, I think.

I think when you mean that when header is hidden, it is displayed, but it's not at a place where it can be seen.